server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -284,72 +292,58 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
+                mapperBuilders.put(mapper.simpleName(), context -> mapper);
                 return this;
             }
 
-            public Builder update(Mapper toMerge, ContentPath contentPath) {
+            public Builder update(FieldMapper toMerge, ContentPath contentPath) {
                 if (mapperBuilders.containsKey(toMerge.simpleName()) == false) {
                     add(toMerge);
                 } else {
-                    Mapper.Builder builder = mapperBuilders.get(toMerge.simpleName());
-                    Mapper existing = builder.build(new BuilderContext(Settings.EMPTY, contentPath));
+                    FieldMapper existing
+                        = mapperBuilders.get(toMerge.simpleName()).apply(new BuilderContext(Settings.EMPTY, contentPath));
                     add(existing.merge(toMerge));
                 }
                 return this;
             }
 
-            @SuppressWarnings("unchecked")
             public MultiFields build(Mapper.Builder mainFieldBuilder, BuilderContext context) {
                 if (mapperBuilders.isEmpty()) {
                     return empty();
                 } else {
+                    Map<String, FieldMapper> mappers = new HashMap<>();
                     context.path().add(mainFieldBuilder.name());
-                    ImmutableOpenMap.Builder mapperBuilders = this.mapperBuilders;
-                    for (ObjectObjectCursor<String, Mapper.Builder> cursor : this.mapperBuilders) {
-                        String key = cursor.key;
-                        Mapper.Builder value = cursor.value;
-                        Mapper mapper = value.build(context);
-                        assert mapper instanceof FieldMapper;
-                        mapperBuilders.put(key, mapper);
+                    for (Map.Entry<String, Function<BuilderContext, FieldMapper>> entry : this.mapperBuilders.entrySet()) {
+                        String key = entry.getKey();
+                        FieldMapper mapper = entry.getValue().apply(context);
+                        mappers.put(key, mapper);
                     }
                     context.path().remove();
-                    ImmutableOpenMap.Builder<String, FieldMapper> mappers = mapperBuilders.cast();
-                    return new MultiFields(mappers.build());
+                    return new MultiFields(mappers);
                 }
             }
         }
 
-        private final ImmutableOpenMap<String, FieldMapper> mappers;
+        private final Map<String, FieldMapper> mappers;
 
-        private MultiFields(ImmutableOpenMap<String, FieldMapper> mappers) {
-            ImmutableOpenMap.Builder<String, FieldMapper> builder = new ImmutableOpenMap.Builder<>();
-            // we disable the all in multi-field mappers
-            for (ObjectObjectCursor<String, FieldMapper> cursor : mappers) {
-                builder.put(cursor.key, cursor.value);
-            }
-            this.mappers = builder.build();
```

Comment:
Weird. That, like, doesn't do anything, right?

Reply:
I guess it used to, hence the comment, but no longer indeed